### PR TITLE
Fixes for striped drawing

### PR DESCRIPTION
**Pull request by bdolman**
First of all, thank you for the great library. This has really helped me make must faster progress on our project than I expected. I identified a few issues around striped drawing of live waveforms and am hoping I can help someone else out by sending these fixes your way.

Lots of detail is in the individual commits, but I wanted to also include a couple of GIFs that illustrate the symptoms of the bugs to help make it clearer.

For 858c157 and d9bf439 ("Handle case where multiple samples are added between draw calls"), I was seeing drawing issues like this once `sampleCount` became greater than what could fit in the window:

multi-sample-add-before

For 0a8def3 ("Fix a blip when doing striped drawing") this is an example of what I was seeing. It's kind of subtle, but if you look closely you'll notice that the amplitude of several of the lines jumps to a different size. This is a result of a different set of samples being used due to the bug.

blip

### CODE REVIEW

**Comment by dmrschmidt** [comment]
Hey @bdolman,

thank you so much for this! I had been noticing this issue on a few rare occasions myself. But I just couldn't figure out where the windowing logic went wrong. So thank you so much for the very detailed commit message explanations. All makes sense now :)

I've just merged your PR and tagged it as a new 9.1.1 release.

**Comment by bdolman** [comment]
I'm glad I could be helpful. Thanks for spending so much time making this library, it's really great!

**Comment by dmrschmidt** [comment]
Thanks so much. If there's any other feedback or improvement suggestions, I'm always interested in hearing them :)

**Comment by bdolman** [comment]
Well @dmrschmidt, there is one thing I'd love your thoughts on. In DSWaveformImage, a waveform initially moves left to right (LEFT IMAGE), but on most other apps these such as iMessage (RIGHT IMAGE), it moves right to left. Our preference for the project I'm working on is the latter. I think I know how to update the code to accomplish that, but I'm wondering what your thoughts are on whether it would be okay to make that the new way that the library works, or if I would be better off adding that as an option when drawing the graph. Happy to do whatever makes the most sense.

**Comment by dmrschmidt** [comment]
That's an interesting question. If I do remember correctly, than the way you described it was, in fact, the way I had originally intended to draw it. I believe I only went with the current form, as that ended up being slightly quicker to finish. So - from my end - there is no particularly strong reason to keep it in the current form.

I'd also argue that the way you described it is a) more "consistent and logical" and b) more what users are used to these days from other apps.

If this this was **some** kind of configuration / different drawing class it would be great in theaory, **but** I also don't necessarily want to make the options too confusing on the other hand.

So in summary, I'd say this is a better new default and I'd just consider this a "backwards incompatible" change for a new 10.0 release.

If you'd like to make this change, I'll be more than happy to merge it. I'll put it on my to do list otherwise and see when I can get to it.

**Comment by bdolman** [comment]
Awesome, thanks @dmrschmidt. I just submitted a PR (https://github.com/dmrschmidt/DSWaveformImage/pull/32) for this. Please feel free to let me know if you prefer to do it a different way or if you have any feedback on it.

**Comment by dmrschmidt** [comment]
Beautiful, thanks a bunch! Merged and released :)
